### ISSUE TITLE
Infrastructure Cost Savings -  WarClub and nodeBB forum migrations

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#2307]
Linode released a 1024 plan a few months ago that cost $5 / month. The new bot servers and test lobby use this server type, we currently have 5 of those machines. We also have a 2048 server ($10 / month) hosting nodeBB and a 4096 server ($20 / month) that is hosting warclub, lobby and some bots.

If we can move nodeBB to a 1024 server we'll save $5 per month, and if we can move lobby and warclub we'll save another $15 / month. WE just got a monthly bill for $55 for servers, lopping $20 off of that would be pretty significant.

**Comment by DanVanAtta** [comment]
Good list @RoiEXLab . I would encourage writing a script to do the forum install so then it's more of a DB move. Anything we touch with forum ideally we update scripts first and then run the scripts.

We may not need to re-install anything, linode provides a re-size option where they will do a data copy from one instance to another. We would be responsible to make sure everything boots up correctly, but at least we don't have to worry about the fact that we don't know how to install much of warclub+xoops (which means we don't really know how to migrate it without quite a lot of effort to reverse engineer the config)

**Comment by RoiEXLab** [comment]
@DanVanAtta 
A point I forgot to mention:
We could of course just make the forum read only at the same time. There are several tools out there, that crawl a website and convert them into a lot of php files.
This way we wouldn't need to worry about any configuration files.
Thoughts?

**Comment by DanVanAtta** [comment]
If we do an image migration, we wouldn't need to worry about config as well, just starting services : )

With that, I'd rather not couple image move with other projects unless needed, we should move to make the forum read-only independently of hosting hardware.

Though this will be an outage for the lobby, so it would be a good time for anything that needs a lobby restart. Lobby code upgrade comes to mind.

**Comment by RoiEXLab** [comment]
Lobby upgrade sounds like a good idea!

**Comment by prastle** [comment]
We can easily lose one of the $5 bot servers to host warclub if it can get image transferred as read only.  We don't need all the bots we have now that summer is over.  It is a long weekend and we still have 8 empty bots during peak est play time.  This is after removing the nj bots on the new forum server from the lobby @RoiEXLab .

**Comment by prastle** [comment]
I am thinking that once test the lobby is finished being used it can host marti by itself ? and maybe a read only warclub? As well as a Source Forge forum read only?  If this is possible on one of the 1024 plans.  This would leave the new forum on the 2048 server that could in an emergency host some extra bots.  Either way ya we can trim this somehow.

**Comment by prastle** [comment]
Another thought @DanVanAtta  is could the test lobby become the new lobby if it was standalone no bots etc? does it have enough memory on that plan?

**Comment by DanVanAtta** [comment]
@prastle to be sure, I am suggesting we drop the current warclub bots and move the image of that server to a smaller and cheaper one. We could move lobby to a dedicated server, though I'm not sure if it is worth having a new server for that, cohosting with warclub I think is reasonable.

> read only warclub? 

I would support that. I somewhat thought it was already done. It' s anew topic though, I opened: https://github.com/triplea-game/triplea/issues/2318 to continue discussing that.

> I am thinking that once test the lobby is finished being used it can host marti by itself ? 

If we remove the warclub bots, we almost get that, xoops + lobby woudl be the only two things on the one server. My plan for the test lobby server was to drop it completely once the lobby has been  upgraded

> As well as a Source Forge forum read only? 

I thought this was done, I'm not sure if we can.. I would support us doing so if possible. I opened #2319 to discuss further.

> Another thought @DanVanAtta is could the test lobby become the new lobby

At the moment I think that would be a complication. We would need to get derby up and running on the test lobby, and that would be a bit painful.

**Comment by prastle** [comment]
So if i follow you correctly we get linode to down size warclub to one of the 1024 plans and lose the bots ? will it have enough memory to be read only and lobby and marti?  Either way yes we can lose the war club bots with all the other ones running.  I do recommend that we keep the new forum on the middle plan tho since it crashed yesterday as roi was moving maps in the already reported glitch.  It obviously needs the ram.  Just my 2 cents good stuff!

**Comment by RoiEXLab** [comment]
@prastle Problem wasn't RAM, but actual Hard Disk Drive Space (all maps together are pretty large...)
If we avoid having a bot "backup", this won't be an issue, which won't be a Problem, if we put the forum alone on a 1024 Server.

**Comment by prastle** [comment]
Yes that was my thought no bots and standalone but a 1024 will handle? cool 👍

**Comment by prastle** [comment]
I just am unsure if losing 13 bots will be to many/  1-7 warclub 10-13 nj

**Comment by DanVanAtta** [comment]
Let's start shutting down warclub bots (@prastle). We have extra bots, if we need more we can spin up another bot server.

**Comment by DanVanAtta** [comment]
10-13 nj should already be uninstalled and halted, 70-73 nj is a different nj server.

**Comment by RoiEXLab** [comment]
10-13 NJ Is currently connected to the test lobby AFAIK

**Comment by DanVanAtta** [comment]
Ah okay. Interesting numbering choice, next time let's use something like  '01 Test' instead of re-using what were the very first NJ bot numbers.

**Comment by prastle** [comment]
@DanVanAtta  You should put a post on forum as well saying they are being closed.  So that other mods don't try and restart them.

**Comment by DanVanAtta** [comment]
Today we moved the nodeBB server to a 1024 plan from 2048, saves $5/ month.

We also moved warclub from 4096 to 2048, which saves $10/ month. We need to reduce the total disk size more before we can move to 1024. 

We current are at 21G of 30G used, I think we can probably clean that up more.

**Comment by DanVanAtta** [comment]
Lobby on 1024 server had all memory except 13M used, so 2048 seems best fit for us for now.

![mem](https://user-images.githubusercontent.com/12397753/30235972-40aa6d6e-94c5-11e7-8a04-62d534038aa0.png)

Closing as we've now the minimum sized servers possible for our configuration.
Now that lobby is on latest, we no longer needed the test lobby server either.

**Comment by RoiEXLab** [comment]
I believe this issue can be closed until we find a solution for  #2318
